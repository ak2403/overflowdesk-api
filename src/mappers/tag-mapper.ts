import { Mapper } from "../types/mappers/mapper";
import { Tag } from "../types/mappers/tag";

export class TagMapper implements Mapper<Tag> {
  private _id!: string;
  private _name!: string;

  set id(id: string) {
    this._id = id;
  }

  set name(name: string) {
    this._name = name;
  }

  build(): Tag {
    return {
      id: this._id,
      name: this._name,
    };
  }
}
