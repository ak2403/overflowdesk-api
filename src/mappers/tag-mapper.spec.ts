import { TagMapper } from "./tag-mapper";

describe("TagMapper()", () => {
  const tagMapper = new TagMapper();

  it("construct tag for passed attribute values", () => {
    tagMapper.id = "123";
    tagMapper.name = "sample-tag";

    expect(tagMapper.build()).toMatchSnapshot();
  });
});
