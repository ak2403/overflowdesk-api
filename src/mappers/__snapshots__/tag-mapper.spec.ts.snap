// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagMapper() construct tag for passed attribute values 1`] = `
{
  "id": "123",
  "name": "sample-tag",
}
`;
